systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
+        List<Secret> secrets = kubeClient(testStorage.getNamespaceName()).listSecrets(testStorage.getNamespaceName()).stream()
+            .filter(secret -> secret.getMetadata().getName().startsWith(testStorage.getClusterName()) && secret.getType().equals("Opaque"))
+            .collect(Collectors.toList());
 
-            verifyPresentLabels(labels, configMap.getMetadata().getLabels());
+        for (Secret secret : secrets) {
+            LOGGER.info("Getting labels from {} secret", secret.getMetadata().getName());
+            verifyAppLabelsForSecretsAndConfigMaps(secret.getMetadata().getLabels());
         }
 
-        LOGGER.info("Waiting for kafka stateful set labels changed {}", labels);
-        StrimziPodSetUtils.waitForStrimziPodSetLabelsChange(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName(), labels);
-
-        LOGGER.info("Verifying kafka labels via stateful set");
-
-        verifyPresentLabels(labels, StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName()));
+        LOGGER.info("---> CONFIG MAPS <---");
 
-        LOGGER.info("Verifying via kafka pods");
-        labels = kubeClient(testStorage.getNamespaceName()).getPod(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0)).getMetadata().getLabels();
+        List<ConfigMap> configMaps = kubeClient(testStorage.getNamespaceName()).listConfigMapsInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName());
 
-        assertThat("Label exists in kafka pods", labelValues[0].equals(labels.get(labelKeys[0])));
-        assertThat("Label exists in kafka pods", labelValues[1].equals(labels.get(labelKeys[1])));
-        assertThat("Label exists in kafka pods", labelValues[2].equals(labels.get(labelKeys[2])));
+        for (ConfigMap configMap : configMaps) {
+            LOGGER.info("Getting labels from {} config map", configMap.getMetadata().getName());
+            verifyAppLabelsForSecretsAndConfigMaps(configMap.getMetadata().getLabels());
+        }
 
-        LOGGER.info("Removing labels: {} -> {}, {} -> {}, {} -> {}", labelKeys[0], labels.get(labelKeys[0]),
-            labelKeys[1], labels.get(labelKeys[1]), labelKeys[2], labels.get(labelKeys[2]));
+        LOGGER.info("---> PVC (both labels and annotation) <---");
 
-        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
-            resource.getMetadata().getLabels().remove(labelKeys[0]);
-            resource.getMetadata().getLabels().remove(labelKeys[1]);
-            resource.getMetadata().getLabels().remove(labelKeys[2]);
-        }, testStorage.getNamespaceName());
+        List<PersistentVolumeClaim> pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```

Comment:
```suggestion
        List<PersistentVolumeClaim> pvcs = kubeClient().listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```
also, when you are listing the PVCs this way, you don't need to filter it afterwards again or?